libs/x-content/src/main/java/org/elasticsearch/common/xcontent/DeprecationHandler.java

```diff
@@ -93,8 +93,36 @@ public void usedDeprecatedField(String parserName, Supplier<XContentLocation> lo
     /**
      * Called when the provided field name matches the current field but the entire
      * field has been marked as deprecated with no replacement
+     * Emits a compatible api warning instead of deprecation warning when isCompatibleDeprecation is true
      * @param usedName the provided field name
      */
     void usedDeprecatedField(String parserName, Supplier<XContentLocation> location, String usedName);
 
+    /**
+     * @see DeprecationHandler#usedDeprecatedName(String, Supplier, String, String)
+     * Emits a compatible api warning instead of deprecation warning when isCompatibleDeprecation is true
+     */
+    default void usedDeprecatedName(String parserName, Supplier<XContentLocation> location, String usedName, String modernName,
```

Comment:
The naming here is a bit confusing.  Can we keep `useDeprecreatedField` and add in `useReplacedField` or someother way besides Name vs. Field to make this a bit more clear ?

Reply:
how about `usedRemovedField` , `usedRenamedField` and `usedReplacedField` ?

Reply:
sorry for being so pedantic... but why `used` to prefix ? would something like this read better ?

LoggingDeprecationHandler.INSTANCE.logRenamedField(null, () -> null, Request.ALLOW_NO_JOBS, Request.ALLOW_NO_MATCH);


Also can the variable/parameter names mirror the method names:

removed : `removedName`
renamed : `oldName` and `currentName`
replaced:  `replacedName`

Reply:
yes, naming is hard :) I agree log* style methods look better for the LoggingDeprecationHandler